src/Init/Data/List/Basic.lean

```diff
@@ -37,11 +38,20 @@ theorem length_add_eq_lengthTRAux (as : List α) (n : Nat) : as.length + n = as.
 @[simp] theorem length_nil : length ([] : List α) = 0 :=
   rfl
 
+/-- Auxiliary for `List.reverse`. `List.reverseAux l r = l.reverse ++ r`, but it is defined directly. -/
 def reverseAux : List α → List α → List α
```

Comment:
Definitely not private, it is used explicitly in both std and mathlib because it's useful for reversing a `DList` (or other isomorphic things), and it has better defeqs than `reverse`. A `where` is fine, although given the aforementioned it might just need a rename to make the standalone use clearer.

Reply:
Right, I forgot to delete this comment